documentation/adr/0018-update-release-notes-every-pr.md

```diff
@@ -0,0 +1,45 @@
+[<-previous](./0017-separate-absolute-and-relative-indexing-APIs-in-pixel-array.md) |
+next->
+
+# 18 - Update Release Notes on Every Pull Request
+
+Date: 2020-Nov-19
+
+## Status
+
+Accepted
+
+## Context
+
+Following the release of v3.5.0 of Horace,
+it was found that breaking changes were not effectively communicated to users.
+A discussion on the best way to communicate these changes,
+as well as the best way for developers to track such changes, was had.
+It was decided release notes provided a solution to this problem.
+
+The workflow for producing release notes needs to be decided.
+
+## Decision
+
+A file is to be kept in the repository to track changes between releases.
+This file is to be updated on every pull request in which:
+
+- an API change is made.
+- a feature is added.
+- a user-facing bug is fixed.
+
+Additions to the file will be written as if addressing users.
+These additions will document behaviour changes,
+as well as changes users will be required to make to their workflows/scripts.
+
+Before every release, this file will be used to compile release notes.
+These release notes will be published alongside releases on GitHub
```

Comment:
Will this be included in the downloadable zips or just on the GitHub?

Reply:
I thought for the time being we'd decided to just publish the docs online i.e. no zip with releases. If there's a need for offline docs, then we can publish a zip along with the release packages.

These lines are actually in disagreement with what's in the SMG, which says that a _summary_ of the release notes goes on the GitHub release and the full notes go in with the user docs.

I'm interested to hear your opinion on that actually. Do we need a summary, or shall we just publish the full set of notes on GitHub as well?

Reply:
Personally, I would say you always want a changelog including, if only to serve as a quick reference to see why all your scripts which used to work are now breaking. While they may also be in the built docs, I would say a changelog is entirely independent from the docs.

Reply:
My thoughts were, the changelog on the GitHub release page should be brief, probably just some bullet point lists: what's new, enhancements, bug fixes. 

The release notes in the user docs can go into more detail, perhaps even including images/screenshots/examples of new features. The GitHub release notes can link to the user docs, so the reader can get more details if they want to.

Reply:
I would say use the same (full) notes for both for ease of updates and giving the users only one place to look for all their needs; less effort for us and less confusion for the user. 

Doing it this way means that the user "has to" read the notes to press download. The user then has no excuse of "oh, I didn't see that because I only looked at the GitHub release", and following a maxim from my old manager "Write code and documentation so as to minimise the number of emails."

I do think other people's input would be useful on this though.

Reply:
I think I'm with you, it's certainly less work for us to just use one document. Probably, if we want to provide more information/examples on new features, that should go in the main user documentation, then the release notes can link to it. That way the documentation for that feature is more permanent.

Reply:
Links to relevant documentation is a very good idea and should definitely be encouraged/explicitly noted.